python/shotgun_desktop/startup.py

```diff
@@ -754,6 +753,7 @@ def main(**kwargs):
 
     from sgtk import authentication
     from sgtk.descriptor import InvalidAppStoreCredentialsError
+    from sgtk.authentication.user import ShotgunSamlUser
```

Comment:
Always import from the `sgtk.authentication` module, not `sgtk.authentication.user`. The public API is anything that can be reached from the `__init__.py files`. Everything else is fair game for us to modify. I haven't looked at the core pull request yet, but if `ShotgunSamlUser` is not brought into `authentication/__init__.py`, please do so like we've done for `ShotgunUser`.